DOCKER.md

```diff
@@ -0,0 +1,123 @@
+# Development with Docker
+
+## Features
+
+* Run development server requirements such as MySQL and Redis in containers.
+
+## Installing Docker
+
+Our Docker development environment requires at least Docker Compose version 2.23 or higher.
+These instructions will walk you through installing a compatible version.
+
+To install Docker, see the appropriate section below:
+[macOS](#macos), [Ubuntu](#ubuntu), [Windows](#windows)
+or consult your package management for your OS.
+
+### macOS
+
+1. Install Docker itself:
+```shell
+brew install docker
+```
+
+1. Install Docker Compose:
+    1. ```shell
+       export DOCKER_CONFIG=${DOCKER_CONFIG:-$HOME/.docker}
+       mkdir -p $DOCKER_CONFIG/cli-plugins
+       curl -SL https://github.com/docker/compose/releases/download/v2.27.0/docker-compose-darwin-$(uname -m) -o $DOCKER_CONFIG/cli-plugins/docker-compose
+       chmod +x $DOCKER_CONFIG/cli-plugins/docker-compose
+       ```
+    1. You can inspect that the version is appropriate by running `docker compose version` and inspecting the result.
+
+1. Install Colima as a container runtime:
+```shell
+brew install colima
```

Comment:
I ran into issues where the MySQL container would terminate randomly (observed consistently when running `bundle exec rake install`) -- we identified it as a "out of memory" issue in [this thread](https://codedotorg.slack.com/archives/C07LFGRN422/p1726176328741309).

It looks like Colima has a 2 GB memory limit by default. Unfortunately, it doesn't look like you pass options like these via `brew services start colima`, which leaves 3 slightly annoying possible recommendations:
1. Open `~/.colima/default/colima.yaml` and edit the memory config.
2. `colima template`, which opens the file to be edited on the command line.
3. run `colima start --memory 4`, which starts colima directly and also appears to modifies the template.

Worth noting that this was only an issue if I had the Selenium containers running at the same time. I'm not sure what the limit should be, but the 2 GB limit enforced by default by Colima is much lower than what Elijah saw on his system (I think Docker is given the system's memory by default).

Reply:
Macs will virtualize the memory for Docker, so you would have to configure Docker (via colima or docker desktop) to pre-allocate a certain amount. The database will consume exactly as much as the ulimit allows (including ALL of it) and that will kill the database service. It is a very annoying problem. It is interesting that the selenium containers are sharing that memory with the database.